qa/sc_evm_test_metamask_related.py

```diff
@@ -0,0 +1,490 @@
+                                         sc_node_configuration)
+        self.sc_nodes_bootstrap_info = bootstrap_sidechain_nodes(self.options, network,
+                                                                 block_timestamp_rewind=720 * 120 * 5,
+                                                                 blockversion=AccountModelBlockVersion)
+
+    def sc_setup_nodes(self):
+        return start_sc_nodes(num_nodes=1, dirname=self.options.tmpdir,
+                              binary=[EVM_APP_BINARY])  # , extra_args=['-agentlib'])
+
+    def run_test(self):
+        # Setting up
+        sc_node = self.sc_nodes[0]
+        mc_block = self.nodes[0].getblock(str(self.sc_nodes_bootstrap_info.mainchain_block_height))
+        mc_block_hex = self.nodes[0].getblock(mc_block["hash"], False)
+        print("SC genesis mc block hex = " + mc_block_hex)
+
+        sc_best_block = sc_node.block_best()["result"]
+
+        assert_equal(sc_best_block["height"], 1, "The best block has not the specified height.")
+
+        # verify MC block reference's inclusion
+        res = is_mainchain_block_included_in_sc_block(sc_best_block["block"], mc_block)
+        assert_true(res, "The mainchain block is not included in SC node.")
+
+        sc_mc_best_block_ref_info = sc_node.mainchain_bestBlockReferenceInfo()["result"]
+        assert_true(
+            check_mainchain_block_reference_info(sc_mc_best_block_ref_info, mc_block),
+            "The mainchain block is not included inside SC block reference info.")
+
+        mc_return_address = self.nodes[0].getnewaddress()
+        # evm_address = generate_account_proposition("seed2", 1)[0]
+
+        ret = sc_node.wallet_createPrivateKeySecp256k1()
+        pprint.pprint(ret)
+        evm_address = format_evm(ret["result"]["proposition"]["address"])
+        print("pubkey = {}".format(evm_address))
+
+        ret = sc_node.wallet_createPrivateKeySecp256k1()
+        other_address = format_evm(ret["result"]["proposition"]["address"])
+
+        ft_amount_in_zen = Decimal("33.22")
+
+        # transfer some fund from MC to SC using the evm address created before
+        forward_transfer_to_sidechain(self.sc_nodes_bootstrap_info.sidechain_id,
+                                      self.nodes[0],
+                                      format_eoa(evm_address),
+                                      ft_amount_in_zen,
+                                      mc_return_address)
+
+        generate_next_block(sc_node, "first node", force_switch_to_next_epoch=True)
+        self.sc_sync_all()
+
+        initial_balance = int(sc_node.rpc_eth_getBalance(str(evm_address), "latest")['result'], 16)
+        transfer_amount = 1
+
+        # EOA transfers via RPC
+
+        eoa_assert_native_balance(sc_node, evm_address, initial_balance)
+        eoa_assert_native_balance(sc_node, other_address, 0)
+
+        eoa_transfer(sc_node, evm_address, other_address, transfer_amount, static_call=True, generate_block=False)
```

Comment:
Expected is no exception here. If the staticcall fails a runtime error will be thrown. This does not return any data, so we can't check the data itself.

Reply:
`static_call=True` means it will do an `eth_call` instead of acutally submitting a transaction, right?
I would expect the same result from both, if the static call fails the tx should too, and vice versa.

Reply:
Yes, that is completely correct. However, since it would throw an exception in the python side, it would end the test there. So there's not really a need to check for anything, since neither should fail in this test